src/quo/components/settings/category/settings/view.cljs

```diff
@@ -15,10 +15,12 @@
        :size   :paragraph-2
        :style  (style/label props)}
       label])
-   [rn/flat-list
-    {:data      data
-     :style     (style/settings-items props)
-     :render-fn settings-item/view
-     :separator [rn/view {:style (style/settings-separator props)}]}]])
+   [rn/view {:style (style/settings-items props)}
+    (for [item data]
+      ^{:key item}
+      [:<>
+       [settings-item/view item]
+       (when-not (= item (last data))
```

Comment:
Just a note @flexsurfer. `last` is inefficient for larger sequences because like a linked list, Clojure has to go one element at a time to find the last one. There's no sort of pointer to the last element. Much faster in larger lists is to grab the index of the last item and compare to that. Also the comparison with `=` would be much faster than comparing `item`, which is a more complex structure.

Take a look at CLJS source code, `last` is expensive.

```clojure
(defn last
  "Return the last item in coll, in linear time"
  [s]
  (let [sn (next s)]
    (if-not (nil? sn)
      (recur sn)
      (first s))))
```

Reply:
We could restrict the type of data to be a vector and use `peek` instead.

Btw, I wonder how expensive the transformation from list to vector is and what is faster:

`(peek (vec data))`
vs
`(last data)`

In case we receive a list.
I guess vec transforms in O(n), but whishing it's not 😅

Reply:
in this current case it's fine i guess because we know there will be a small number of items, and i don't think it's super critical here, just don't like indexes in clojure code :)

Reply:
Interesting question @ulisesmac, I don't know either without benchmarks.

In general, we should just not call `last` inside a loop of a collection, it's quite inefficient, unless we know it's a vector as you hinted.

I think just calling `(-> data count dec)` is significantly cheaper than trying to convert to a vector and repeatedly call `last` inside a loop because then we can just do a number comparison of `(= index last-index)`, which is probably as fast as we can go to identify the last element.

Reply:
> In general, we should just not call last inside a loop of a collection, it's quite inefficient, unless we know it's a vector as you hinted.

I agree, we could just extract that calculation outside the `for` loop.